lib/url.c

```diff
@@ -2908,7 +2908,8 @@ static bool IsPipeliningPossible(const struct Curl_easy *handle,
                                  const struct connectdata *conn)
 {
   /* If a HTTP protocol and pipelining is enabled */
-  if(conn->handler->protocol & PROTO_FAMILY_HTTP) {
+  if(!conn->bits.close &&
+      (conn->handler->protocol & PROTO_FAMILY_HTTP)) {
```

Comment:
please review [code style column alignment](https://github.com/curl/curl/blob/master/docs/CODE_STYLE.md#column-alignment), you need to back it up one  see the other if statements in the function for example. also it could fit on one line, for example
~~~c
  if((conn->handler->protocol & PROTO_FAMILY_HTTP) && !conn->bits.close) {
~~~